.github/CONTRIBUTING.md

```diff
@@ -1,17 +1,177 @@
+- **Describe the behavior you observed after following the steps** and point out what exactly is the problem with that behavior.
+- **Explain which behavior you expected to see instead and why.**
+- **Include screenshots or animated GIFs** which show you following the described steps and clearly demonstrate the problem. If you use the keyboard while following the steps, use [this tool](https://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and [this tool](https://github.com/colinkeenan/silentcast) or [this tool](https://gitlab.gnome.org/Archive/byzanz) on Linux.
+
+## 🛠 Suggesting Enhancements
+
+Feature requests are tracked as [GitHub issues](https://guides.github.com/features/issues/). Create an issue on that repository and provide the following information:
+
+- **Use a clear and descriptive title** for the issue to identify the suggestion.
+- **Provide a in detail description of the suggested enhancement** in as many details as possible.
+- **Explain why this enhancement would be useful** to the project and the community.
+
+## 📝 Cloning the project & creating PR
+
+### Fork the repository.
+
+Click on the fork button on the top of the page. This will create a copy of this repository in your account. Instead click [here](https://github.com/aidenybai/million/fork) to fork the repository.
+
+### Clone the forked repository.
+
+```bash
+ git clone https://github.com/<your-username>/million.git
+```
+
+or if use the Github CLI
+
+```bash
+gh repo clone <your-username>/million
+```
+
+### Navigate to the project directory.
+
+```bash
+cd million
+```
+
+### Create a new branch (naming convention: type-description-issueNo)
+
+Kindly give your branch a more descriptive name like `docs-contributing-guide-2` instead of `patch-1`.
+
+You could follow this convention. Some ideas to get you started:
+
+- Feature Updates: `feat-<brief 2-3 words-Description>-<ISSUE_NO>`
+- Bug Fixes: `fix-<brief 2-3 words-Description>-<ISSUE_NO>`
+- Documentation: `docs-<brief 2-3 words-Description>-<ISSUE_NO>`
+- And so on...
+
+To create a new branch, use the following command:
+
+```bash
+git checkout -b your-branch-name
+```
+
+### Make the necessary changes.
+
+### Stage your changes and commit.
+
+```bash
+git add . # Stages all the changes
+git commit -m "<your_commit_message>"
+```
+
+Your commit message should be something which gives concise idea of the issue you are solving.
+
+We implement the Conventional Commits specification for commit messages. This specification is a lightweight convention on top of commit messages. It provides an easy set of rules for creating an explicit commit history; which makes it easier to write automated tools on top of. This convention dovetails with SemVer, by describing the features, fixes, and breaking changes made in commit messages.
+
+The commit message should be structured as follows:
+
+```
+<type>(optional scope): <description>
+```
+
+Push your local commits to the remote repository.
+
+```bash
```

Comment:
ask the user to run `pnpm cleanup`, `pnpm lint`, `pnpm test before pushing`

Reply:
Done